blog/_posts/2021-10-19-scala-3.1.0-released.md

```diff
@@ -0,0 +1,113 @@
+
+You don't need to worry about having to migrate your projects to the new version. Scala 3 has strong compatibility guarantees, and all code working in 3.0.2 will also work in 3.1. On the other hand, switching to Scala 3.1 allows you to use many improvements and a handful of newly stabilized APIs.
+
+## What's new in 3.1
+
+### New experimental feature: safer exceptions
+
+A new experimental feature that allows declaring and checking which exceptions can be thrown. It relies on the effects as implicit capabilities pattern.
+
+You can opt-in using following import.
+
+```scala
+import language.experimental.saferExceptions
+```
+
+Now it is possible to mark types with exception that cna be thrown during the evalutaion:
+
+```scala
+def f(x: Double): Double canThrow LimitExceeded =
+  if x < limit then f(x) else throw LimitExceeded())
+```
+
+You can read more in [the document proposing this feature](https://github.com/lampepfl/dotty/blob/release-3.1.0/docs/docs/reference/experimental/canthrow.md).
+
+### Efficient bytecode for matching over strings
+
+When Scala 3.0 has been compiling pattern matches with `Int` as a scrutinee type, it was emitting bytecode containing very efficient instructions: `lookupswitch` or `tableswitch`. Now in Scala 3.1, similar behavior was implemented for matching over `String` literals.
+
+```scala
+fruit match
+  case "apple"  => 1
+  case "orange" => 2
+  case "banana" => 3
+  case _        => 0
+```
+
+will now generate `lookupswitch` over hash codes of `String` literals instead of a chain of conditional expressions. This behavior is consistent with what Java is doing and more performant than the old approach.
+
+### Support for `-Wconf` and `@nowarn`
+
+We have added a `-Wconf` compiler flag that allows filtering and configuring compiler warnings (silence them, or turn them into errors).
+
+We have also integrated [the silencer plugin](https://github.com/ghik/silencer) into the compiler, which allows suppressing warnings locally using the `@nowarn` annotation.
+
+`-Wconf` and `@nowarn` work largely the same way as in Scala 2, but some filters for selecting warnings are different. For the more details see the output of `-Wconf:help` flag.
+
+### Simplified Manifest synthesis
+
+For the sake of compatibility with Scala 2, the Scala 3 compiler now generates given instances for the `Manifest` trait. The new implementation is a slightly simplified approximation of the Scala 2 implementation. It guarantees that any of the expressions:
+
+- `manifest[A] == manifest[B]`
+- `manifest[A].runtimeClass == manifest[B].runtimeClass`
+- `optManifest[A] == optManifest[B]`
+- `optManifest[A].asInstanceOf[ClassTag[A]].runtimeClass == optManifest[B].asInstanceOf[ClassTag[B]].runtimeClass`
+
+that was true in Scala 2, will also remain true when compiled with Scala 3.
+
+### Other changes
```

Comment:
Mention the upgrade to Scala.js 1.7.x here? It impacts users since they will need to upgrade their Scala.js version to 1.7.0+ before they can upgrade to Scala 3.1.

Reply:
Can you please suggest the change? I don't know much about Scala.js and don't want to include anything that can be misleading for the users.